test/e2e/reconciler/cache/replication_test.go

```diff
@@ -224,11 +228,10 @@ func replicateShardScenario(ctx context.Context, t *testing.T, server framework.
 		kcpShardClusterDynamicClient,
 		cacheKcpClusterDynamicClient,
 		core.RootCluster,
-		"test-shard",
 		"Shard",
 		corev1alpha1.SchemeGroupVersion.WithResource("shards"),
 		&corev1alpha1.Shard{
-			ObjectMeta: metav1.ObjectMeta{Name: "test-shard"},
+			ObjectMeta: metav1.ObjectMeta{Name: withPseudoRandomSuffix("test-shard")},
```

Comment:
nit: can all or most of these use GenerateName?

Reply:
yes, we could do that but that would require some/bigger changes because the tests need stable name so that they can find resources for modification/deletion.

I can prepare a follow-up PR for that^